force_wfmanager/wfmanager_plugin.py

```diff
@@ -1,35 +1,77 @@
 from envisage.api import Plugin
 from envisage.ui.tasks.api import TaskFactory
-
+from envisage.ui.tasks.task_extension import TaskExtension
+from pyface.action.action import Action
+from pyface.action.action_item import ActionItem
+from pyface.tasks.action.schema_addition import SchemaAddition
 from traits.api import List
 
-from force_bdss.factory_registry_plugin import FACTORY_REGISTRY_PLUGIN_ID
-
-from force_wfmanager.wfmanager_task import WfManagerTask
+from force_wfmanager.wfmanager_results_task import WfManagerResultsTask
+from force_wfmanager.wfmanager_setup_task import WfManagerSetupTask
+from force_bdss.api import FACTORY_REGISTRY_PLUGIN_ID
 
 
 class WfManagerPlugin(Plugin):
     """ The basic WfManager """
 
     TASKS = 'envisage.ui.tasks.tasks'
 
+    TASK_EXTENSIONS = 'envisage.ui.tasks.task_extensions'
+
     id = 'force_wfmanager.wfmanager_plugin'
     name = 'Workflow Manager'
-    workflow_file = None
 
     tasks = List(contributes_to=TASKS)
+    task_extensions = List(contributes_to=TASK_EXTENSIONS)
+
+    def __init__(self, workflow_file):
+        super(WfManagerPlugin, self).__init__()
+        self.workflow_file = workflow_file
 
     def _tasks_default(self):
-        return [TaskFactory(id='force_wfmanager.wfmanager_task',
-                            name='Workflow Manager',
-                            factory=self._create_task)]
+        return [TaskFactory(id='force_wfmanager.wfmanager_setup_task',
+                            name='Workflow Manager (Setup)',
+                            factory=self._create_setup_task),
+                TaskFactory(id='force_wfmanager.wfmanager_results_task',
+                            name='Workflow Manager (Results)',
+                            factory=self._create_results_task)
+                ]
+
+    def _task_extensions_default(self):
+        """Extensions (toolbars, menubars etc.) to be added to a Task.
+        If the TaskExtension has no task_id, the Extension applies to
+        all Tasks"""
+        return [TaskExtension(
+            actions=[SchemaAddition(
+                factory=self._exit_action,
+                path='MenuBar/File')]
+            )
+        ]
 
-    def _create_task(self):
+    def _create_setup_task(self):
         factory_registry = self.application.get_plugin(
             FACTORY_REGISTRY_PLUGIN_ID)
-
-        wf_manager_task = WfManagerTask(factory_registry=factory_registry)
+        wf_manager_setup_task = WfManagerSetupTask(
+            factory_registry=factory_registry,
+        )
         if self.workflow_file is not None:
-            wf_manager_task.open_workflow_file(self.workflow_file)
+            wf_manager_setup_task.open_workflow_file(self.workflow_file)
+
+        return wf_manager_setup_task
+
+    def _create_results_task(self):
+        factory_registry = self.application.get_plugin(
```

Comment:
You should decide which convention you use for ending parenthesis. Line 64 adds no line break while line 66 has a line break. I typically add a line break. I suggest you check what the standard is in the force repositories.